nilearn/masking.py

```diff
@@ -566,15 +567,20 @@ def compute_gray_matter_mask(target_img, threshold=.5,
     mask : nibabel.Nifti1Image
         The brain mask (3D image)
     """
+
     return compute_brain_mask(target_img=target_img, threshold=threshold,
                               connected=connected, opening=opening,
-                              memory=memory, verbose=verbose)
+                              memory=memory, verbose=verbose,
+                              mask_type='whole-brain',
+                              resolution=2)
 
 
-def compute_brain_mask(target_img, threshold=.5, connected=True,
-                       opening=2, memory=None, verbose=0):
-    """Compute the whole-brain mask. This mask is calculated through the
-    resampling of the MNI152 template mask onto the target image.
+def compute_brain_mask(target_img, threshold=.5, connected=True, opening=2,
+                       memory=None, verbose=0, mask_type='whole-brain',
+                       resolution=2):
```

Comment:
i think it's a bit confusing to ask for a resolution as the template gets resampled to the `target_img` anyway. I would suggest always loading the 1mm template and resampling it to `target_img`, and removing the `resolution` parameter

Reply:
OK. I did this change. I also agree it makes more sense if we keep the 1mm-template (regardless whether we fetch it or load it directly from the nilearn package). Yet, this change it is causing some tests to fail, e.g. `nilearn/input_data/tests/test_multi_nifti_masker.py::test_compute_multi_gray_matter_mask`. In this test function, the params `mask` and `mask2` become zero arrays, causing lines 191 and 192 to fail. Changing the erosion doesn't help. What should I do here?

Reply:
the test is right to fail because there is a breaking change to `compute_multi_gray_matter_mask`: it used to return a whole-brain mask and now it is suddenly returning a gray matter mask. the situation is exactly the same as for `compute_gray_matter_mask`: we need to deprecate `compute_multi_gray_matter_mask` (but keep it for now with a deprecation warning) add a `compute_multi_brain_mask` with a `mask_type` parameter.

also I think the maskers should continue using a whole-brain mask when `strategy="template"`

besides this bigger issue,
- the resampling done in the test is a bit extreme because a large image is resampled to a very small one, maybe the threshold is a bit high
- the expected image is hard-coded, it may be the case that a few voxels differ when using a different template so you may have to adjust the expected result
but I would fix the masker issue first